test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -447,6 +456,1909 @@ public async Task GetInstalledVersion_WithAssetsFile_ChangingPackageSpec_Returns
+                            target, packageB.Id, uninstallationContext, testNuGetProjectContext, CancellationToken.None);
+
+                        results.AddRange(resolvedActions.Select(a => new ResolvedAction(target, a)));
+                    }
+
+                    var actions = results.Select(a => a.Action).ToArray();
+
+                    await nuGetPackageManager.ExecuteNuGetProjectActionsAsync(
+                        netCorePackageReferenceProjects,
+                        actions,
+                        testNuGetProjectContext,
+                        new SourceCacheContext(),
+                        CancellationToken.None);
+
+                    // Assert
+                    Assert.Equal(initialInstalledPackages.Count(), 2);
+                    var builtIntegratedActions = actions.OfType<BuildIntegratedProjectAction>().ToList();
+                    Assert.Equal(actions.Count(), targetProjects .Count());
+                    Assert.Equal(actions.Count(), builtIntegratedActions.Count());
+                    Assert.True(builtIntegratedActions.All(b => b.RestoreResult.Success));
+                    var restoringLogs = testNuGetProjectContext.Logs.Value.Where(l => l.StartsWith("Restoring packages for ")).ToList();
+                    var restoredLogs = testNuGetProjectContext.Logs.Value.Where(l => l.StartsWith("Restored ")).ToList();
+                    // Making sure project1 restored only once, not many.
+                    // https://github.com/NuGet/Home/issues/9932
+                    Assert.Equal(restoringLogs.Count(l => l.EndsWith("project1.csproj...")), 1);
+                    Assert.Equal(restoredLogs.Count(l => l.Contains("project1.csproj")), 1);
+                    // Making sure project2 restored only once, not many. 
+                    Assert.Equal(restoringLogs.Count(l => l.EndsWith("project2.csproj...")), 1);
+                    Assert.Equal(restoredLogs.Count(l => l.Contains("project2.csproj")), 1);
+                    var writingAssetsLogs = testNuGetProjectContext.Logs.Value.Where(l => l.StartsWith("Writing assets file to disk.")).ToList();
+                    // Only 2 write to assets for above 2 projects, never more than that.
+                    Assert.Equal(writingAssetsLogs.Count, 2);
+                    // There should be no warning/error.
+                    Assert.Equal(builtIntegratedActions.Sum(b => b.RestoreResult.LogMessages.Count()), 0);
+                }
+            }
+            finally
+            {
+                foreach (TestDirectory projectDirectory in projectDirectories)
+                {
+                    projectDirectory.Dispose();
+                }
+            }
+        }
+
+        private TestProjectJsonBuildIntegratedNuGetProject CreateTestNetCorePackageReferenceProject(string projectName, string projectFullPath, ProjectSystemCache projectSystemCache, TestProjectSystemServices projectServices = null)
```

Comment:
Actually this is a PackageReference based project, just the name is bad.

Reply:
Good catch. Rename to `TestNetCorePackageReferenceProject`.